### PR TITLE
crawl

### PR DESCRIPTION
**Pull request by xu1jia2qi3**

### CODE REVIEW

**Comment by xu1jia2qi3** [comment]
add crawler

File: crawl.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+async def fetch_image_by_id(url, img_id):
+    async with aiohttp.ClientSession(connector=aiohttp.TCPConnector(verify_ssl=False)) as session:
+        async with session.get(url) as response:
+            try:
+                image_data = await response.content.read()
+                save_folder = './Snapshot/'
+                createFolder(f'{save_folder}{img_id}/')
+                with open(f'{save_folder}{img_id}/{img_id}.jpg', 'wb') as f:
+                    f.write(image_data)
+                # print(f'Download complete - {img_id}.jpg')
```
Comment (tiddler) [inline crawl.py]: We can keep the `print` as log.

File: crawl.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            except Exception as e:
+                print(f'Download failed - {url}', e)
+
+
+async def fetch_all_images():
+    sem = asyncio.Semaphore(10)
```
Comment (tiddler) [inline crawl.py]: Change the number `10` here as a config variable, like `MAX_DOWNLOAD_NUMBER`.

File: crawl.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    try:
+        with open('./camera.json', 'r') as f:
```
Comment (tiddler) [inline crawl.py]: Same, change the path to a config variable, like `CAMERA_DATA_PATH`.

**Review by tiddler — CHANGES_REQUESTED** [review]
Please fix some issues.

**Comment by tiddler** [comment]
Also, create a folder, like `src` to place all code files.

**Comment by tiddler** [comment]
@xu1jia2qi3 Any updates?

**Comment by xu1jia2qi3** [comment]
fixed code

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "C:\\Users\\DD\\AppData\\Local\\Programs\\Python\\Python36\\python.exe"
```
Comment (tiddler) [inline .vscode/settings.json]: Unnecessary file.

File: .gitignore [hunk 5]
```diff
@@ -2,6 +2,7 @@
 __pycache__/
 *.py[cod]
 *$py.class
+*.docx
```
Comment (tiddler) [inline .gitignore]: Add `.vscode/` as well.

File: crawl.py [hunk 6]
```diff
@@ -14,20 +17,19 @@ def createFolder(path):
...
             try:
-                image_data = await response.content.read()
-                save_folder = './Snapshot/'
-                createFolder(f'{save_folder}{img_id}/')
-                with open(f'{save_folder}{img_id}/{img_id}.jpg', 'wb') as f:
+                image_data = await response.content.read()                
+                createFolder(f'{SAVE_FOLDER}{img_id}/')
+                with open(f'{SAVE_FOLDER}{img_id}/{img_id}.jpg', 'wb') as f:
                     f.write(image_data)
                 # print(f'Download complete - {img_id}.jpg')
```
Comment (tiddler) [inline crawl.py]: Uncomment this line.

**Review by tiddler — CHANGES_REQUESTED** [review]
Remove jpg files?

File: .gitignore [hunk 7]
```diff
@@ -2,6 +2,9 @@
...
+.vscode/
+.Snapshot/
```
Comment (tiddler) [inline .gitignore]: Change `.Snapshot` to `Snapshot`

**Review by tiddler — CHANGES_REQUESTED** [review]

**Review by tiddler — APPROVED** [review]
Good Job.

**Pull request merged**